app/controllers/users_controller.rb

```diff
@@ -143,16 +143,16 @@ def update
 
       fields = UserField.all
       fields = fields.where(editable: true) unless current_user.staff?
-      fields.each do |f|
-        field_id = f.id.to_s
+      fields.each do |field|
+        field_id = field.id.to_s
         next unless params[:user_fields].has_key?(field_id)
 
-        val = params[:user_fields][field_id]
-        val = nil if val === "false"
-        val = val[0...UserField.max_length] if val
+        value = clean_custom_field_values(field)
+        value = nil if value === "false"
+        value = value[0...UserField.max_length] if value
```

Comment:
I'm wondering if all this logic should be moved into `clean_custom_field_values`

Reply:
I think we could, although the method would be slightly longer. I'm going to leave it here for simplicity, for now.